modules/gitea.py

```diff
@@ -0,0 +1,239 @@
+        elif gitea_event == "pull_request_review":
+            outputs = self.pull_request_review(event, full_name, data)
+        elif gitea_event == "pull_request_review_comment":
+            outputs = self.pull_request_review_comment(event, full_name, data)
+        elif gitea_event == "issue_comment":
+            outputs = self.issue_comment(event, full_name, data)
+        elif gitea_event == "issues":
+            outputs = self.issues(event, full_name, data)
+        elif gitea_event == "create":
+            outputs = self.create(event, full_name, data)
+        elif gitea_event == "delete":
+            outputs = self.delete(event, full_name, data)
+        elif gitea_event == "release":
+            outputs = self.release(event, full_name, data)
+
+        if outputs:
+            for server, channel in targets:
+                for output in outputs:
+                    output = "(%s) %s" % (full_name, output)
+                    self.events.on("send.stdout").call(target=channel,
+                        module_name="Gitea", server=server, message=output,
+                        hide_prefix=channel.get_setting(
+                        "gitea-hide-prefix", False))
+
+        return True
+
+    def _change_count(self, n, symbol, color):
+        return utils.irc.color("%s%d" % (symbol, n), color)+utils.irc.bold("")
+    def _added(self, n):
+        return self._change_count(n, "+", utils.consts.GREEN)
+    def _removed(self, n):
+        return self._change_count(n, "-", utils.consts.RED)
+    def _modified(self, n):
+        return self._change_count(n, "~", utils.consts.PURPLE)
+
+    def _short_hash(self, hash):
+        return hash[:8]
+
+    def _flat_unique(self, commits, key):
+        return set(itertools.chain(*(commit[key] for commit in commits)))
+
+    def push(self, event, full_name, data):
+        outputs = []
+        branch = data["ref"].split("/", 2)[2]
+        branch = utils.irc.color(branch, utils.consts.LIGHTBLUE)
+
+        if len(data["commits"]) <= 3:
+            for commit in data["commits"]:
+                id = self._short_hash(commit["id"])
+                message = commit["message"].split("\n")[0].strip()
+                author = commit["author"]["name"] or commit["author"]["login"]
+                author = utils.irc.bold(author)
+                url = self.COMMIT_URL % (full_name, id)
+
+                outputs.append(("Commit by {author} to {branch}: {msg} - {url}"
+                               ).format(author=author, branch=branch, msg=message, url=url))
+        else:
+            first_id = self._short_hash(data["before"])
+            last_id = self._short_hash(data["commits"][-1]["id"])
+            pusher = utils.irc.bold(data["pusher"]["name"])
+            url = self.COMMIT_RANGE_URL % (full_name, first_id, last_id)
+
+            commits = data["commits"]
+            added = self._added(len(self._flat_unique(commits, "added")))
```

Comment:
`added`/`removed`/`modified` has been removed for individual commits. surely it should also be removed here?